service/controller/clusterapi/v28/key/cluster.go

```diff
@@ -62,7 +62,7 @@ func BucketObjectName(cluster v1alpha1.Cluster, role string) string {
 }
 
 func ClusterAPIEndpoint(cluster v1alpha1.Cluster) string {
-	return fmt.Sprintf("api.%s.%s", ClusterID(cluster), ClusterBaseDomain(cluster))
+	return fmt.Sprintf("api.%s.k8s.%s", ClusterID(cluster), ClusterBaseDomain(cluster))
```

Comment:
In `Cluster` CR there is only base domain present while `AWSConfig` had different zones etc. Therefore this needed the additional `k8s.` DNS label injection. Same applies for those two other endpoints below.

Reply:
I do not understand this. I thought the base domain contains the `k8s` part already? Why do we add this magic string now everywhere again?

Reply:
I don't think the base domain has ever contained that. It was my false assumption earlier, but here is an example from live `AWSConfig`:
```
      hostedZones:                                                                                                                                             
        api:                     
          name: gauss.eu-central-1.aws.gigantic.io                                                                                                             
        etcd:                                                                                                                                                  
          name: gauss.eu-central-1.aws.gigantic.io                                                                                                             
        ingress:                                                                                                                                               
          name: gauss.eu-central-1.aws.gigantic.io 
```

Rest of the stuff is then added on the fly.

Reply:
What I mean is that you added the `k8s` prefix to the base domain with the Cluster API types. The example in the godoc also shows this. See https://godoc.org/github.com/giantswarm/apiextensions/pkg/apis/cluster/v1alpha1#AWSClusterSpec. Why do we need to add the prefix here when it is already in the base domain? How does the CR look like with which you tested?

Reply:
Yeah, as I wrote, it was my false assumption earlier that base domain contained that, but that's not the case, so we need to remove it from that example too. My CRs are these:
**Cluster**
```
apiVersion: cluster.k8s.io/v1alpha1
kind: Cluster
metadata:
 labels:                                                                       
   giantswarm.io/cluster: "8y5kc"                                              
   giantswarm.io/customer: "giantswarm" 
   giantswarm.io/organization: "giantswarm"
   aws-operator.giantswarm.io/version: 6.2.0
   cluster-operator.giantswarm.io/version: 0.17.0
   release.giantswarm.io/version: 7.3.1 
 name: 8y5kc 
spec:               
 providerSpec:
   value:  
     apiVersion: cluster.giantswarm.io/v1alpha1
     kind: AWSClusterSpec
     metadata:
       name: 8y5kc                                                             
     cluster:                                                                  
       description: my fancy cluster
       dns:            
         domain: gauss.eu-central-1.aws.gigantic.io
       versionBundle:                                                          
         version: 6.2.0
     provider:                       
       credentialSecret:              
         name: credential-default
         namespace: giantswarm
       master:      
         availabilityZone: eu-central-1a
         instanceType: m4.large
       region: eu-central-1
```

**MachineDeployment**
```
apiVersion: cluster.k8s.io/v1alpha1                                            
kind: MachineDeployment
metadata:
  labels:
    aws-operator.giantswarm.io/version: 6.2.0
    cluster-operator.giantswarm.io/version: 0.17.0
    giantswarm.io/cluster: 8y5kc
    giantswarm.io/customer: giantswarm
    giantswarm.io/organization: "giantswarm"
    giantswarm.io/machine-deployment: al9qy
    release.giantswarm.io/version: 7.3.1
  name: al9qy
  namespace: default
spec:
  template:
    spec:
      providerSpec:
        value:
          apiVersion: cluster.giantswarm.io/v1alpha1
          kind: AWSMachineDeploymentSpec
          metadata:
            name: al9qy
          nodePool:
            description: my fancy node pool
            machine:
              dockerVolumeSizeGB: 100
              kubeletVolumeSizeGB: 100
            scaling:
              max: 3
              min: 3
          provider:
            availabilityZones:
            - eu-central-1b
            worker:
              instanceType: m4.xlarge
```

Reply:
Can you then please provide a PR to fix the example in the godoc?

Reply:
For the record, this was merged here: https://github.com/giantswarm/apiextensions/pull/253